pkg/curatedpackages/reader.go

```diff
@@ -53,29 +64,46 @@ func (r *PackageReader) ReadImagesFromBundles(ctx context.Context, b *releasev1.
 		images = append(images, packageImages...)
 	}
 
-	return images, err
+	return removeDuplicateImages(images), err
 }
 
 func (r *PackageReader) ReadChartsFromBundles(ctx context.Context, b *releasev1.Bundles) []releasev1.Image {
-	images := r.ManifestReader.ReadChartsFromBundles(ctx, b)
+	var images []releasev1.Image
+	eksaCharts := append(images, r.ManifestReader.ReadChartsFromBundles(ctx, b)...)
+	if r.AllImages {
+		images = append(images, eksaCharts...)
+	}
+
+	// Generate bundle repository
+	for _, chart := range eksaCharts {
+		if strings.Contains(chart.Repository(), "eks-anywhere-packages") {
+			var bundleRepository releasev1.Image
+			chart.DeepCopyInto(&bundleRepository)
+			bundleRepository.URI = strings.Replace(bundleRepository.URI, "eks-anywhere-packages", "eks-anywhere-packages-bundles", 1)
+			bundleRepository.URI = strings.Replace(bundleRepository.URI, bundleRepository.Version(), "v1-24-latest", 1)
+			bundleRepository.ImageDigest = ""
```

Comment:
Minor: this feels like a nice function to extract.

Reply:
Another time perhaps.